examples/graphiql-plugin.js

```diff
@@ -0,0 +1,115 @@
+
+const pluginSrc = `
+(function (global, factory) {
+  typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('react')) :
+  typeof define === 'function' && define.amd ? define(['exports', 'react'], factory) :
+  (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global.samplePlugin = {}, global.React));
+})(this, (function (exports, React) { 'use strict';
+  function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }
+  var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
+  class SampleDataManager extends EventTarget {
+    constructor() { super(); this.sampleData = []; }
+    getExplainData() { return this.sampleData }
+    setExplain(sampleData) { this.sampleData = sampleData || []; this.dispatchEvent(new Event('updateSampleData')); }
+  }
+
+  const sampleDataManager = new SampleDataManager();
+  const useSampleData = () => {
+    const [sampleData, setSampleData] = React.useState(sampleDataManager.getExplainData());
+    React.useEffect(() => {
+      const eventListener = sampleDataManager.addEventListener(
+        'updateSampleData', (e, value) => {setSampleData(_ => e.target.sampleData || []);});
+      return () => {sampleDataManager.removeEventListener('updateSampleData', eventListener);}
+    }, []);
+    return {sampleData}
+  };
+
+  function Content() {
+    const { sampleData } = useSampleData();
+    return (
+      React__default["default"].createElement('div', {style: { maxWidth: '300px'}}, [
+        React__default["default"].createElement('div', {style: { height: '100%'}}, ["This is a sample plugin"]),
+        sampleData && React__default["default"].createElement('pre', null, [JSON.stringify(sampleData, null, 2)])
+      ]))}
+
+  function Icon() {return React__default["default"].createElement('p', {style: { width: '100%', display: 'flex', justifyContent: 'center'}}, ["P"])}
+  function graphiqlSamplePlugin() {return umdPlugin()}
+  
+  /*
+   * The plugin should have \`umdPlugin\` function that exports:
+   *   title: string. The title of the plugin
+   *   icon: React component. The icon shown in the toolbar
+   *   content: React component.
+   */  
+  function umdPlugin() {
+    return { 
+      title: 'GraphiQL Sample',
+      icon: () => Icon(),
+      content: () => { return Content() }
+    }
+  }
+
+  /*
+   * A function to intercept the fetch data can be exported to allow access to the data.
+   * Graphiql do not provide an easy way to access the retrieved data.
+   */  
+  function parseFetchResponse(data) {
+    if (data.data) {sampleDataManager.setExplain(data.data);} // Do something with the data
+    return data // It should return the data
+  }
+  exports.graphiqlSamplePlugin = graphiqlSamplePlugin;
+  exports.parseFetchResponse = parseFetchResponse;
+  exports.umdPlugin = umdPlugin;
+  Object.defineProperty(exports, '__esModule', { value: true });
+}));
+`
```

Comment:
How could I generate a plugin like this?

Reply:
you can use rollup
`const packageJson = require('./package.json')

const rollup = [
  {
    input: 'src/export.js',
    output: [
      {
        file: packageJson.umd,
        format: 'umd',
        sourcemap: true,
        name: ''
      }
    ],
    external: ['react', '@graphiql/toolkit'],
    plugins: [
      resolve({
        extensions: ['.js', '.jsx']
      }),
      svgr(),
      commonjs(),
      jsx({ factory: 'React.createElement' })
    ]
  }`

Reply:
The plugin is generated creating a UMD file with rollup.

The project with the plugin we are building can be found here:

https://github.com/nearform/mercurius-explain-graphiql-plugin (the docs are still in progress but will be there very soon)

The plugin should export a structure in this format:

```
{ 
  title: 'GraphiQL Sample',
  icon: () => <Icon/>,
  content: () => { return <Content /> }
}
```

Rollup should export it in `umd` format:

```
    output: [
      {
        file: packageJson.umd,
        format: 'umd',
        sourcemap: true,
        name: 'mercuriusExplain' // the name of the export variable, available in the graphiql/main.js using window.mercuriusExplain
      }
    ],
```

I've a repository with a scaffolding to create `graphiql` plugins, I need to clean it up and add the docs, it can be a starting point to create graphiql plugins.

Reply:
Can you document this? It seems non-obvious.

Reply:
@mcollina I've added a document with the instructions to create a basic plugin:

https://github.com/mercurius-js/mercurius/pull/893/files#diff-c13c46bc9c5f536ae3a38dec17a0bb8570a80affb8a7e9f200eae30c45199747